deeppavlov/models/embedders/cove_embedder.py

```diff
@@ -0,0 +1,150 @@
+                num_layers,
+                num_units,
+                tf.estimator.ModeKeys.PREDICT,
+                residual_connections=residual_connections,
+                cell_class=cell_class)
+
+            cell_bw = build_cell(
+                num_layers,
+                num_units,
+                tf.estimator.ModeKeys.PREDICT,
+                residual_connections=residual_connections,
+                cell_class=cell_class)
+
+            encoder_outputs_tup, encoder_state_tup = tf.nn.bidirectional_dynamic_rnn(
+                cell_fw,
+                cell_bw,
+                inputs,
+                sequence_length=self.sequence_length,
+                dtype=inputs.dtype)
+
+        with tf.variable_scope('', reuse=tf.AUTO_REUSE):
+            tf_vars = []
+            tf_names = []
+            for name, value in checkpoint_values.items():
+                if 'bidirectional_rnn' in name:
+                    tf_vars.append(tf.get_variable(name))
+                    tf_names.append(name)
+            placeholders = [tf.placeholder(v.dtype, shape=v.shape) for v in tf_vars]
+            assign_ops = [tf.assign(v, p) for (v, p) in zip(tf_vars, placeholders)]
+
+        reducer = ConcatReducer()
+
+        # merge (by concatenation) backward and forward rnn_cells' outputs/states
+        self.encoder_outputs = reducer.zip_and_reduce(encoder_outputs_tup[0], encoder_outputs_tup[1])
+        self.encoder_state = reducer.zip_and_reduce(encoder_state_tup[0], encoder_state_tup[1])
+
+        self.sess = tf.Session()
+        self.sess.run(tf.global_variables_initializer())
+        self.sess.run(tf.tables_initializer())
+        self.sess.run(assign_ops, {p: checkpoint_values[name] for p, name in zip(placeholders, tf_names)})
+
+    def _load_checkpoint_values(self, *args, **kwargs):
+        """
+        Load weights from checkpoint
+        """
+
+        # loading checkpoint values:
+        var_list = tf.train.list_variables(self.checkpoint_path)
+        values = {}
+        for name, shape in var_list:
+            if not name.startswith("global_step"):
+                values[name] = np.zeros(shape)
+        reader = tf.train.load_checkpoint(self.checkpoint_path)
+        for name in values:
+            values[name] += reader.get_tensor(name)
+        return values
+
+    @overrides
+    def __call__(self, batch: List[List[str]], lengths: List[int], *args, **kwargs):
+        """
+        Embed data
+        """
+        outputs = self._encode(batch, lengths)
+        encoded = None
+
+        if self.reduce_method == 'mean':
+            encoded = np.mean(outputs, axis=1)
+        elif self.reduce_method == 'sum':
+            encoded = np.sum(outputs, axis=1)
+        elif self.reduce_method == 'max':
+            encoded = np.max(outputs, axis=1)
+        elif self.reduce_method == 'none':
+            encoded = outputs
+        else:
+            log.error("None of reducing methods has been defined")
```

Comment:
Can this be resolved in `__init__`? You can make `self.reduce_method` a method there and then just call
```
encoded = self.reduce_method(outputs)
```
or something like that

Reply:
Yes, agree with it